File: frontend/src/types/dailyNote.ts
export interface DailyNote {
  id: number;
  project_id: number;
  title: string;
  note_date?: string | null;
  content: string;
  completed_tasks?: string | null;
  blockers?: string | null;
  next_steps?: string | null;
  ai_summary?: string | null;
  created_at: string;
  updated_at: string;
}

export interface DailyNoteCreatePayload {
  title: string;
  note_date?: string | null;
  content: string;
  completed_tasks?: string | null;
  blockers?: string | null;
  next_steps?: string | null;
  ai_summary?: string | null;
}

export interface DailyNoteUpdatePayload {
  title?: string;
  note_date?: string | null;
  content?: string;
  completed_tasks?: string | null;
  blockers?: string | null;
  next_steps?: string | null;
  ai_summary?: string | null;
}

File: backend/app/models/daily_note.py
from sqlalchemy import Column, Date, DateTime, ForeignKey, Integer, String, Text
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func

from app.db.database import Base


class DailyNote(Base):
    __tablename__ = "daily_notes"

    id = Column(Integer, primary_key=True, index=True)
    project_id = Column(
        Integer,
        ForeignKey("projects.id", ondelete="CASCADE"),
        nullable=False,
        index=True,
    )
    title = Column(String(200), nullable=False)
    note_date = Column(Date, nullable=True)
    content = Column(Text, nullable=False)
    completed_tasks = Column(Text, nullable=True)
    blockers = Column(Text, nullable=True)
    next_steps = Column(Text, nullable=True)
    ai_summary = Column(Text, nullable=True)
    created_at = Column(DateTime(timezone=True), server_default=func.now(), nullable=False)
    updated_at = Column(
        DateTime(timezone=True),
        server_default=func.now(),
        onupdate=func.now(),
        nullable=False,
    )

    project = relationship("Project", back_populates="daily_notes")
